build.gradle

```diff
@@ -79,29 +80,13 @@ pluginBundle {
     }
 }
 
-task releaseNeeded {
-    ext.needed = true
-    doLast {
-        def pullRequest = System.getenv('TRAVIS_PULL_REQUEST')?.matches("\\d+")
-        def onMaster = "master" == System.getenv('TRAVIS_BRANCH')
-        ext.needed = !pullRequest && onMaster
-        if (needed) {
-            logger.lifecycle("{} - Release to Gradle Plugin Portal is needed.", path)
-        } else {
-            logger.lifecycle("{} - Current branch is not 'master' or we're building pull request.\n" +
-                    "  Release to Gradle Plugin Portal is not needed.", path)
-        }
-    }
-}
-
-//TODO ww add -Preleasing.dryRun=false to commandLine when releaseNeeded takes into account [ci skipRelease] in commit msg
 task travisRelease {
     dependsOn releaseNeeded
-    onlyIf { releaseNeeded.needed }
+    onlyIf { !releaseNeeded.releaseNotNeeded }
```

Comment:
I suggest we use 'assertReleaseNeeded' in Travis CI instead of onlyIf (similar to how it is set up for mockito and example project). It makes things a bit simpler complexity wise, it makes things slightly easier to deploy/test locally, if you need to test/publish from local.